src/MRS1000_node.cpp

```diff
@@ -0,0 +1,234 @@
+
+  n.param<std::string>("host", host, "192.168.1.2");
+  n.param<std::string>("frame_id", frame_id, "laser");
+  n.param<int>("port", port, 2111);
+
+  cloud.header.frame_id = frame_id;
+  cloud.header.stamp = ros::Time::now();
+  cloud.height = 4; // 4 layers
+  cloud.width = 275*4+1; // 275° with a resolution of 0.25° (+ 1)
+
+  //Fill the fields using the PointCloudModifier
+  sensor_msgs::PointCloud2Modifier modifier(cloud);
+  modifier.setPointCloud2Fields(4,
+    "x", 1, sensor_msgs::PointField::FLOAT32,
+    "y", 1, sensor_msgs::PointField::FLOAT32,
+    "z", 1, sensor_msgs::PointField::FLOAT32,
+    "intensity", 1, sensor_msgs::PointField::FLOAT32);
+  //modifier.setPointCloud2FieldsByString(2, "xyz", "intensity");
+  cloud.is_bigendian = false;
+  cloud.is_dense = false;
+
+  while (ros::ok())
+  {
+    ROS_INFO_STREAM("Connecting to laser at " << host);
+    laser.connect(host, port);
+    if (!laser.isConnected())
+    {
+      ROS_WARN("Unable to connect, retrying.");
+      ros::Duration(1).sleep();
+      continue;
+    }
+
+    ROS_DEBUG("Logging in to laser.");
+    laser.login();
+    /* in the evaluation prototype, those functions are not (yet) available
+    cfg = laser.getScanCfg();
+    outputRange = laser.getScanOutputRange();
+
+    if (cfg.scaningFrequency != 5000)
+    {
+      laser.disconnect();
+      ROS_WARN("Unable to get laser output range. Retrying.");
+      ros::Duration(1).sleep();
+      continue;
+    }*/
+
+    ROS_INFO("Connected to laser.");
+
+    /* in the evaluation prototype, those functions are not (yet) available
+    dataCfg.outputChannel = 1;
+    dataCfg.remission = true;
+    dataCfg.resolution = 1;
+    dataCfg.encoder = 0;
+    dataCfg.position = false;
+    dataCfg.deviceName = false;
+    dataCfg.outputInterval = 1;
+
+    ROS_DEBUG("Setting scan data configuration.");
+    laser.setScanDataCfg(dataCfg);
+    */
+    /* in the evaluation prototype, those functions are not (yet) available
+     * TODO: How does this work? Do I need to always start using SOPAS?
+    ROS_DEBUG("Starting measurements.");
+    laser.startMeas();
+
+    ROS_DEBUG("Waiting for ready status.");
+    ros::Time ready_status_timeout = ros::Time::now() + ros::Duration(5);
+
+    //while(1)
+    //{
+    status_t stat = laser.queryStatus();
+    ros::Duration(1.0).sleep();
+    if (stat != ready_for_measurement)
+    {
+      ROS_WARN("Laser not ready. Retrying initialization.");
+      laser.disconnect();
+      ros::Duration(1).sleep();
+      continue;
+    }
+    /*if (stat == ready_for_measurement)
+    {
+      ROS_DEBUG("Ready status achieved.");
+      break;
+    }
+
+      if (ros::Time::now() > ready_status_timeout)
+      {
+        ROS_WARN("Timed out waiting for ready status. Trying again.");
+        laser.disconnect();
+        continue;
+      }
+
+      if (!ros::ok())
+      {
+        laser.disconnect();
+        return 1;
+      }
+    }*/
```

Comment:
what about this comment?
if some functionalities are not yet available, why not comment them in the main loop used below?